### PR TITLE
backend for upload

### PR DESCRIPTION
**Pull request by darmhoo**
#### What does this PR do?
- (replace me) Explain WHY this PR is required. E.g.: What feature it adds, what problem it solves, …

#### Description of Task proposed in this pull request?
- (replace me) A description of WHAT changed in the codebase. Additionally add the reasoning if it's complex or abstract. 

#### How should this be manually tested (Quality Assurance)?
- Explain how to manually test this feature. For example if changes were made in
the UI or in the API, explain where and if any specific access is needed.

#### What are the relevant pivotal tracker stories?
- [Story ID:  replace text with story id](paste Pivotal Tracker link here)

#### Any background context you want to add (Operations Impact)?
- (replace me) List the steps that must be taken for this PR to work.
E.g.: `npm install`, `Add "github_key" to environment variables, ...`

Be sure to include all systems that needs to be changed or which system is
affected by the change (Ex: Requires Styled Components to be installed and
configured in `babel.rc`, file to be updated).

#### What I have learned working on this feature: [If you don't put anything here, you are doing it wrong!]
- (replace me) I learned how to create a pull request template

### CODE REVIEW

File: app/Http/Controllers/NewDocumentController.php [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
```
Comment (harmlessprince) [inline app/Http/Controllers/NewDocumentController.php:43]: How are we tracking the document type and name

File: database/migrations/new/2023_03_01_013336_create_new_documents_table.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+class CreateNewDocumentsTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('new_documents', function (Blueprint $table) {
```
Comment (harmlessprince) [inline database/migrations/new/2023_03_01_013336_create_new_documents_table.php:16]: We need document name and document type filled

e.g 

document_type = guarantor_id_card 

document name = "Guarnator ID Card"

you can easily set document type from the document name supplied when the document is been created

Comment (harmlessprince) [reply database/migrations/new/2023_03_01_013336_create_new_documents_table.php:16]: You can have a class that holds this, where key is the document type 
```
<?php

namespace App\Utils;

class DocumentType
{
    const GUARANTOR_ID_CARD =  [
        'key' => 'guarantor_id_card',
        'name' => 'Guarantor ID Card'
    ];

    const GOODS_IN_TRANSIT_INSURANCE = [
        'key' => 'goods_in_transit_insurance',
        'name' => 'Goods In Transit Insurance'
    ];
    const FIDELITY_INSURANCE = [
        'key' => 'fidelity_insurance',
        'name' => 'Fidelity Insurance'
    ];

    const CAC_DOCUMENT = [
        'key' => 'cac_document',
        'name' => 'CAC Document'
    ];


    const TRUCK_PICTURE = [
        'key' => 'truck_picture',
        'name' => 'Truck Picture'
    ];
    const TRUCK_PROOF_OF_OWNERSHIP = [
        'key' => 'truck_proof_of_ownership',
        'name' => 'Truck Proof of Ownership'
    ];
    const TRUCK_ROAD_WORTHINESS = [
        'key' => 'truck_road_worthiness',
        'name' => 'Truck Road Worthiness'
    ];

    const TRUCK_LICENSE = [
        'key' => 'truck_license',
        'name' => 'Truck License'
    ];
    const TRUCK_INSURANCE = [
        'key' => 'truck_insurance',
        'name' => 'Truck Insurance'
    ];
}
```

File: app/Customer.php [hunk 3]
```diff
@@ -242,4 +242,8 @@ public function latestAmortizationPayed()
     public function recommendation(){
         return $this->hasMany(Recommendation::class);
     }
+
+    public function newDocuments(){
```
[the remark this answers is not in the document]
Comment (darmhoo) [reply app/Customer.php:246]: There is documents already. that we are actively using

**Review by teebarg — APPROVED** [review]

**Pull request merged**